volume/drivers/adapter.go

```diff
@@ -24,6 +24,11 @@ func (a *volumeDriverAdapter) Name() string {
 }
 
 func (a *volumeDriverAdapter) Create(name string, opts map[string]string) (volume.Volume, error) {
+	v, err := a.proxy.Get(name)
+	if err == nil && v.DriverName() != a.Name() {
+		return nil, errors.New("A volume named " + name + " already exists with the " + v.DriverName() + " driver. Choose a different volume name.")
+	}
+
```

Comment:
I wonder if this is already handled; looking at https://github.com/moby/moby/blob/8e14f278c4de7acb65d40cdfefb54a5cb4d6eb97/volume/service/store.go#L511-L524

And, more specifically, these lines https://github.com/moby/moby/blob/8e14f278c4de7acb65d40cdfefb54a5cb4d6eb97/volume/service/store.go#L553-L558